### PR TITLE
Scalar mul for MatZ

### PR DESCRIPTION
**Pull request by Marcel583**
Implements scalar multiplication for MatZ.

### CODE REVIEW

**Review by jnsiemer — CHANGES_REQUESTED** [review]
Good work, please consider my comments below to use scalar multiplication more natively, keep the code to maintain as small as possible, and keep the doc comments consistent/correct.

File: src/integer/mat_z/arithmetic/mul_scalar.rs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! Implementation of scalar multiplication for [`MatZ`] matrices.
+
+use super::super::MatZ;
+use crate::integer::Z;
+use crate::macros::arithmetics::{
+    arithmetic_trait_borrowed_to_owned, arithmetic_trait_mixed_borrowed_owned,
+};
+use crate::traits::*;
```
[the remark this answers is not in the document]
Comment (Marcel583) [reply src/integer/mat_z/arithmetic/mul_scalar.rs:16]: They are imported in compile time, not run time, therefore it is not more efficient to import them separately.
Furthermore, if we change the traits, we do not need to refactor later (safes time in implementation).

[the remark this answers is not in the document]
Comment (jnsiemer) [reply src/integer/mat_z/arithmetic/mul_scalar.rs:16]: Compile time sounds just a little bit less aweful than runtime as all doc tests are compiled on their own. Hence, I would love to also optimize compile time (if it's as easy as it is here). It's aweful enough that we do `use qfall_math::traits::*;` for these functions implementing our traits, but this way, we will import them all for all doc tests that we have (if this isn't optimized in another way, but then please show me a source that it doesn't impact compile time of doc tests that do not themselves explicitely import these traits).

File: src/integer/mat_z/arithmetic/mul_scalar.rs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+use flint_sys::fmpz_mat::fmpz_mat_scalar_mul_fmpz;
+use std::ops::Mul;
+
+impl Mul<&Z> for &MatZ {
```
[the remark this answers is not in the document]
Comment (Marcel583) [reply src/integer/mat_z/arithmetic/mul_scalar.rs:20]: It is not possible simply with Into<Z>, but I have added a macro for that.

File: src/integer/mat_z/arithmetic/mul_scalar.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Marcel583) [reply src/integer/mat_z/arithmetic/mul_scalar.rs:17]: Ok

**Review by jnsiemer — APPROVED** [review]

**Pull request merged**